tests/disaster-recovery/mdr/test_active_hub_down_and_restore.py

```diff
@@ -0,0 +1,282 @@
+        """
+
+        def finalizer():
+            if (
+                self.primary_cluster_name
+                and get_fence_state(
+                    drcluster_name=self.primary_cluster_name,
+                    switch_ctx=get_passive_acm_index(),
+                )
+                == "Fenced"
+            ):
+                enable_unfence(
+                    drcluster_name=self.primary_cluster_name,
+                    switch_ctx=get_passive_acm_index(),
+                )
+                gracefully_reboot_ocp_nodes(self.namespace, self.primary_cluster_name)
+
+        request.addfinalizer(finalizer)
+
+    @pytest.mark.polarion_id("OCS-XXXX")
+    def test_application_failover_and_relocate(
+        self,
+        setup_acm_ui,
+        nodes_multicluster,
+        dr_workload,
+    ):
+
+        """
+        Tests to verify failover and relocate all apps in a single zone after a zone disruption
+
+        """
+
+        if config.RUN.get("mdr_failover_via_ui"):
+            ocs_version = version.get_semantic_ocs_version_from_config()
+            if ocs_version <= version.VERSION_4_12:
+                logger.error(
+                    "ODF/ACM version isn't supported for Failover/Relocate operation"
+                )
+                raise NotImplementedError
+
+        acm_obj = AcmAddClusters()
+        # ToDO: Create appset and multiple apps
+        workload = dr_workload(num_of_subscription=1)[0]
+        self.namespace = workload.workload_namespace
+
+        # Create application on Primary managed cluster
+        set_current_primary_cluster_context(workload.workload_namespace)
+        self.primary_cluster_name = get_current_primary_cluster_name(
+            namespace=workload.workload_namespace
+        )
+        secondary_cluster_name = get_current_secondary_cluster_name(
+            workload.workload_namespace
+        )
+
+        # Create backup-schedule on active hub
+        create_backup_schedule()
+        # ToDo: To verfiy all the backups are taken
+        wait_time = 300
+        logger.info(f"Wait {wait_time} until backup is taken ")
+        time.sleep(wait_time)
+
+        # Install gitops operator on passive hub
+        dep_obj = Deployment()
+        dep_obj.deploy_gitops_operator(switch_ctx=get_passive_acm_index())
+
+        # Get the active hub nodes
+        config.switch_ctx(get_active_acm_index())
+        active_hub_index = config.cur_index
+        active_hub_cluster_node_objs = get_node_objs()
+
+        # Shutdown active hub nodes
+        logger.info("Shutting down all the nodes of active hub")
+        nodes_multicluster[active_hub_index].stop_nodes(active_hub_cluster_node_objs)
+        logger.info(
+            "All nodes of active hub zone are powered off, "
+            f"wait {wait_time} seconds before restoring in passive hub"
+        )
+
+        # Restore new hub
+        restore_backup()
+        logger.info(f"Wait {wait_time} until restores are taken ")
```

Comment:
Post restore, status of Velero might be important as a verification step

Reply:
Agreed, and added verification to check is restore succeeded